### PR TITLE
[webgpu] Update INT_DIV

### PR DESCRIPTION
**Pull request by hujiajie**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

### CODE REVIEW

**Review by qjia7 — APPROVED** [review]
LGTM, thanks!

File: tfjs-backend-webgpu/src/binary_op_util.ts [hunk 1]
```diff
@@ -66,36 +66,14 @@ const GREATER_EQUAL = `
...
-  let s = sign(a) * sign(b);
-  let ia = i32(round(a));
-  let ib = i32(round(b));
-  return f32(idiv(ia, ib, s));
+  let numerator = round(a);
+  let denominator = round(b);
+  let remainder = round(numerator % denominator);
+  let quotient = round((numerator - remainder) / denominator);
+  let resultTemp =
+      select(quotient, quotient - 1, sign(remainder) == -sign(denominator));
```
Comment (gyagp) [inline tfjs-backend-webgpu/src/binary_op_util.ts:75]: Several questions and comments:
1) round() in WGSL has a special definition: "Result is the integer k nearest to e, as a floating point value. When e lies halfway between integers k and k + 1, the result is k when k is even, and k + 1 when k is odd." Will this introduce a diff between CPU and WebGPU solution?
2) For "let remainder = round(numerator % denominator);" Is round() necessary?
3) What's the result of "9 / (-2)" in this code?
4) numerator -> dividend, denominator -> divisor?
5) Please ensure we have enough unit test cases for them, especially when dividend and/or divisor are halfway between integers, the results from "(numerator - remainder) / denominator" are halfway between integers, etc.

Comment (hujiajie) [reply tfjs-backend-webgpu/src/binary_op_util.ts:75]: > Several questions and comments:
> 
> 1. round() in WGSL has a special definition: "Result is the integer k nearest to e, as a floating point value. When e lies halfway between integers k and k + 1, the result is k when k is even, and k + 1 when k is odd." Will this introduce a diff between CPU and WebGPU solution?

This is not clearly mentioned in the TensorFlow.js API doc. Even if the CPU rounding mode differs from the GPU backend, I cannot tell which one is more correct, or if either is fine.

> 2. For "let remainder = round(numerator % denominator);" Is round() necessary?

No, but I feel that's more aligned with neighboring lines.

> 3. What's the result of "9 / (-2)" in this code?

-5

> 4. numerator -> dividend, denominator -> divisor?

The API doc prefers numerator/denominator.

> 5. Please ensure we have enough unit test cases for them, especially when dividend and/or divisor are halfway between integers, the results from "(numerator - remainder) / denominator" are halfway between integers, etc.

Comment (gyagp) [reply tfjs-backend-webgpu/src/binary_op_util.ts:75]: Different languages may have different results.
[python]
(-9) % 2 -> 1
9 % (-2)  -> -1

[JavaScript]
(-9) % 2 -> -1
9 % (-2)  -> 1

So your result will be at least different from the JS's result. 
With round(), we introduce another level of difference between CPU and GPU, at least for the quotient calculation.

Comment (hujiajie) [reply tfjs-backend-webgpu/src/binary_op_util.ts:75]: Please note `INT_DIV` is exclusively used by `floorDiv()`, and the doc states the result should be floored. This matches with the current CPU behavior (`Math.floor(a / b)`).

Comment (qjia7) [reply tfjs-backend-webgpu/src/binary_op_util.ts:75]: JiaJie, if `INT_DIV` is only used for `floorDiv`, will it be better to rename `INT_DIV` to `FLOOR_DIV`?

**Comment by jzm-intel** [comment]
I have concern that the whole implementation of `INT_DIV` and `floorDiv` may conflict their expected semantic. The implementation, before and after this PR, is not coherent with `floor(x/y)`, which is the definition of `floorDiv(x, y)`. It only works correctly for integer input, and actually only tested on `int32`. 
For example, input `(5.9, 2)` will result in `6.0 / 2.0 = 3.0` instead of `floor(5.9 / 2.0) = floor(2.95) = 2.0`, and input `(3.7, 1.9)` will result in `4.0 / 2.0 = 2.0` instead of `floor(3.7 / 1.9) = floor(1.947368...) = 1.0`.
I think the `round` of two inputs in original implementation are intend for handling integer inputs passing to WebGL as float, and we should revisit them.

**Review by qjia7 — COMMENTED** [review]
Good work, Jiajie. Please also add some test cases for `float32` type in tfjs-core.

**Review by gyagp — APPROVED** [review]
I defer to Zhaoming's review on this.

**Comment by qjia7** [comment]
@jzm-intel Do you have any concern with the latest change?

File: tfjs-backend-webgpu/src/kernels/FloorDiv.ts [hunk 2]
```diff
@@ -21,9 +21,11 @@ import {BinaryOpType} from '../binary_op_util';
 import {binaryKernelFunc} from '../kernel_utils/kernel_funcs_utils';
 import {floorDivImplCPU} from '../kernel_utils/shared';
 
-export const floorDiv =
-    binaryKernelFunc({opType: BinaryOpType.INT_DIV,
-                      cpuKernelImpl: floorDivImplCPU, dtype: 'int32'});
+export const floorDiv = binaryKernelFunc({
+  opType: BinaryOpType.FLOOR_DIV,
+  cpuKernelImpl: floorDivImplCPU,
+  dtype: 'int32'
```
Comment (jzm-intel) [inline tfjs-backend-webgpu/src/kernels/FloorDiv.ts:27]: Why does it state as 'int32'? This is the last thing that confuse me....

**Review by jzm-intel — APPROVED** [review]
LGTM, thanks

File: tfjs-backend-webgpu/src/kernels/FloorDiv.ts (hunk 2 above)
Comment (hujiajie) [reply tfjs-backend-webgpu/src/kernels/FloorDiv.ts:27]: This is the result dtype.

Comment (jzm-intel) [reply tfjs-backend-webgpu/src/kernels/FloorDiv.ts:27]: Thanks, still LGTM.

**Pull request merged**